### PR TITLE
RDM-6955 Introduce a mechanism to detect duplicate object IDs and fai…

### PR DESCRIPTION
**Pull request by andrewfolga**
…l framework with exception. Refactor JsonStoreWithInheritance for better readability - break down to smaller methods, rename method and variable names.

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDM-6955

### Change description ###
Duplicate Object IDs Not Reported

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by andrewfolga** [comment]
@MSancaktutar Which file do you mean? I could not find a comment on any of the files?

File: src/test/resources/framework-test-data-duplicate-guids/features-with-duplicate-guids/common/response/header/Common_Response_Headers.td.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+	"_guid_": "Common_Response_Headers",
+
+	"X-Frame-Options": "DENY",
+	"Cache-Control": "no-cache, no-store, max-age=0, must-revalidate",
+	"X-Content-Type-Options": "nosniff",
+	"Pragma": "no-cache",
+	"Expires": "0",
+	"X-XSS-Protection": "1; mode=block",
+	"Date": "[[ANY]]"
+}
```
Comment (MSancaktutar) [inline src/test/resources/framework-test-data-duplicate-guids/features-with-duplicate-guids/common/response/header/Common_Response_Headers.td.json]: eol at eof

File: src/test/resources/framework-test-data/features/F-000/S-000.td.json [hunk 2]
```diff
@@ -16,7 +16,6 @@
     }
   },
   "expectedResponse": {
-    "responseCode": 200,
```
Comment (MSancaktutar) [inline src/test/resources/framework-test-data/features/F-000/S-000.td.json]: why remove this?

File: src/main/java/uk/gov/hmcts/jsonstore/JsonStoreWithInheritance.java [hunk 3]
```diff
@@ -1,215 +1,220 @@
...
+        }
+
+        Iterator<JsonNode> fields = node.iterator();
+        while (fields.hasNext())
+            removeInheritanceMechanismFields(fields.next());
+    }
+
+    protected abstract void buildObjectStore() throws Exception;
+
+    protected void validateGUIDs(String guid) {
```
Comment (MSancaktutar) [inline src/main/java/uk/gov/hmcts/jsonstore/JsonStoreWithInheritance.java]: validateGuid --> singular name is better as it's validating the 1 guid from the method param.

Comment (andrewfolga) [reply src/main/java/uk/gov/hmcts/jsonstore/JsonStoreWithInheritance.java]: Done now

File: src/test/java/uk/gov/hmcts/jsonstore/JsonStoreWithInheritanceTest.java [hunk 4]
```diff
@@ -18,6 +18,15 @@
     private static final HttpTestDataSource TEST_DATA_RESOURCE = new JsonStoreHttpTestDataSource(
             TEST_DATA_RESOURCE_PACKAGES);
 
+    private static final String[] TEST_DATA_RESOURCE_PACKAGES_WITH_DUPLICATE_GUIDS = {"framework-test-data-duplicate-guids/features-with-duplicate-guids"};
+    private static final HttpTestDataSource TEST_DATA_RESOURCE_WITH_DUPLICATE_GUIDS = new JsonStoreHttpTestDataSource(
+            TEST_DATA_RESOURCE_PACKAGES_WITH_DUPLICATE_GUIDS);
+
+    @Test(expected = RuntimeException.class)
+    public void shouldHaveOnlyUniqueGUID() {
+        TEST_DATA_RESOURCE_WITH_DUPLICATE_GUIDS.getDataForTestCall("F-050_Test_Data_Base");
```
Comment (andrewfolga) [inline src/test/java/uk/gov/hmcts/jsonstore/JsonStoreWithInheritanceTest.java]: Check exception message.

Comment (andrewfolga) [reply src/test/java/uk/gov/hmcts/jsonstore/JsonStoreWithInheritanceTest.java]: e.g. BrokenTestDataException

File: src/main/java/uk/gov/hmcts/befta/exception/InvalidTestDataException.java [hunk 5]
```diff
@@ -0,0 +1,14 @@
+package uk.gov.hmcts.befta.exception;
+
+public class InvalidTestDataException extends RuntimeException {
```
Comment (MSancaktutar) [inline src/main/java/uk/gov/hmcts/befta/exception/InvalidTestDataException.java]: Could this please extend FunctionalTestException?

File: src/test/resources/framework-test-data-duplicate-guids/features-with-duplicate-guids/common/request/user/Common_User_For_Request.td.json [hunk 6]
```diff
@@ -0,0 +1,6 @@
+{
+	"_guid_": "Common_User_For_Request",
+
+	"username": "[[$CCD_CASEWORKER_AUTOTEST_EMAIL]]",
+	"password": "[[$CCD_CASEWORKER_AUTOTEST_PASSWORD]]"
+}
```
Comment (MSancaktutar) [inline src/test/resources/framework-test-data-duplicate-guids/features-with-duplicate-guids/common/request/user/Common_User_For_Request.td.json]: eol at eof

File: src/main/java/uk/gov/hmcts/befta/exception/InvalidTestDataException.java (hunk 5 above)
Comment (andrewfolga) [reply src/main/java/uk/gov/hmcts/befta/exception/InvalidTestDataException.java]: Sure. Done.

File: src/test/resources/framework-test-data-duplicate-guids/features-with-duplicate-guids/common/request/user/Common_User_For_Request.td.json (hunk 6 above)
Comment (andrewfolga) [reply src/test/resources/framework-test-data-duplicate-guids/features-with-duplicate-guids/common/request/user/Common_User_For_Request.td.json]: Done.

**Pull request merged**